sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -6843,9 +6562,8 @@ object functions {
    * @since 3.0.0
    */
   // scalastyle:on line.size.limit
-  def schema_of_json(json: Column, options: java.util.Map[String, String]): Column = {
-    withExpr(SchemaOfJson(json.expr, options.asScala.toMap))
-  }
+  def schema_of_json(json: Column, options: java.util.Map[String, String]): Column =
```

Comment:
I think these should be the few exceptions that we should still keep them as they were, because the expression (`SchemaOfJson` in this case) takes non-expression inputs, and it's tricky to define a protocol that can pass non-expression inputs as expressions.

Reply:
No sure I get your point. `SchemaOfJson` seems to accept options as expressions: https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala#L776-L778. We just follow Connect here.

Reply:
Oh I see, then I'm fine with it. It's inevitable that we need to create map expression from the language native (scala or python) map values.